### PR TITLE
Handle Previously Uncaught Exceptions

### PR DESCRIPTION
**Pull request by kingthorin**
Added try/catch block around sendAndReceive for exceptions identified
via zapbot automated wavsep testing.

Fixes zaproxy/zaproxy#2178
Fixes zaproxy/zaproxy#2179

### CODE REVIEW

File: src/org/zaproxy/zap/extension/ascanrulesBeta/SQLInjectionMySQL.java [hunk 1]
```diff
@@ -196,6 +197,10 @@ public void scan(HttpMessage originalMessage, String paramName, String originalP
 			catch (java.net.SocketTimeoutException e) {
 				//to be expected occasionally, if the base query was one that contains some parameters exploiting time based SQL injection?
 				if ( this.debugEnabled ) log.debug("The Base Time Check timed out on ["+msgTimeBaseline.getRequestHeader().getMethod()+"] URL ["+msgTimeBaseline.getRequestHeader().getURI().getURI()+"]");
+			} catch (SocketException ex) {
+				log.info("Caught " + ex.getClass().getName() + " " + ex.getMessage() +
```
Comment (psiinon) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/SQLInjectionMySQL.java]: I think INFO logging should be used only for occasional 'significant' events, so vote for these sort of events should be DEBUG. Recording stats on them is another matter - that would be really useful, but out of scope for this change.

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/SQLInjectionMySQL.java]: Ok, I'll tweak it and submit a fixup.

Comment (kingthorin) [inline src/org/zaproxy/zap/extension/ascanrulesBeta/SQLInjectionMySQL.java]: Should I also tweak the previous few to debug vs info as well? For consistency.

https://github.com/zaproxy/zap-extensions/pull/228
https://github.com/zaproxy/zap-extensions/pull/221

**Comment by kingthorin** [comment]
Tweaked.

I'll work on aligning https://github.com/zaproxy/zap-extensions/pull/228 & https://github.com/zaproxy/zap-extensions/pull/221 separately.

**Comment by psiinon** [comment]
Is it deliberate that we return after a SocketException but not after a SocketTimeoutException?

**Comment by thc202** [comment]
Looks good to me.

**Comment by psiinon** [comment]
Ah, thc202 explained it to me ;)

**Pull request merged**